backend/src/main/java/org/eclipse/tractusx/puris/backend/masterdata/logic/service/MaterialPartnerRelationServiceImpl.java

```diff
@@ -76,12 +85,67 @@ public MaterialPartnerRelation create(MaterialPartnerRelation materialPartnerRel
         if (searchResult == null) {
             dtmService.update(materialPartnerRelation);
             executorService.submit(new DtrRegistrationTask(materialPartnerRelation, "CREATE", 3));
+            if (materialPartnerRelation.getMaterial().isMaterialFlag() && materialPartnerRelation.isPartnerSuppliesMaterial()
+                && materialPartnerRelation.getPartnerCXNumber() == null) {
+                log.info("Attempting CX-Id fetch for Material " + materialPartnerRelation.getMaterial().getOwnMaterialNumber() +
+                    " from Supplier-Partner " + materialPartnerRelation.getPartner().getBpnl());
+                executorService.submit(new PartTypeInformationRetrievalTask(materialPartnerRelation, 3));
+            }
             return mprRepository.save(materialPartnerRelation);
         }
         log.error("Could not create MaterialPartnerRelation, " + materialPartnerRelation.getKey() + " already exists");
         return null;
     }
 
+    @AllArgsConstructor
+    private class PartTypeInformationRetrievalTask implements Callable<Boolean> {
+        final MaterialPartnerRelation materialPartnerRelation;
+        int retries;
+
+        @Override
+        public Boolean call() throws Exception {
+            Thread.sleep(100);
+            if (retries < 0) {
+                log.warn("PartTypeInformation fetch from " + materialPartnerRelation.getPartner().getBpnl() +
+                    " for " + materialPartnerRelation.getMaterial().getOwnMaterialNumber() + " failed");
+                return false;
+            }
+            String[] data = edcAdapterService.getContractForPartTypeInfoApi(materialPartnerRelation.getPartner());
+            if (data != null) {
+                String authKey = data[0];
+                String authCode = data[1];
+                String endpoint = data[2];
+                var response = edcAdapterService.getProxyPullRequest(endpoint, authKey, authCode,
```

Comment:
I would argue that it is technically impossible that the endpoint from the EDR-Dto diverges from anything, that could be retreived from an AAS. If it would -against all expectations- diverge then that would be an indicator that the url from the AAS is outdated ... but we are not expecting that to happen, right?

Reply:
Exactly. The href in the AAS / SubmodelDescriptor should remain stable. At least we're assuming that.